File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int maxDiff = 0;
        for(int i = 0; i<colors.length; i++){
            for(int j = 0; j<colors.length; j++){
                if(i != j && colors[i] != colors[j]){
                    int tempDiff = Math.abs(i - j);
                    maxDiff = Math.max(tempDiff, maxDiff);
                }
            }
        }
        return maxDiff;
    }
}

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.js
/**
 * @param {number[]} colors
 * @return {number}
 */
var maxDistance = function(colors) {
    
    
    let len = colors.length, start = 0, end = len -1
    
    while(colors[0] == colors[end]) end--
    while(colors[len-1] == colors[start]) start++
    
    return Math.max(len-1-start, end)
    
};